install/kots/manifests/gitpod-kots-config.yaml

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+# Licensed under the MIT License. See License-MIT.txt in the project root for license information.
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: gitpod-kots-config
+  labels:
+    app: gitpod
+    component: gitpod-installer
+data:
+  # General settings
+  CURSOR: repl{{ Cursor | quote }}
```

Comment:
:bulb: (idea; non-blocking) Asking Replicated for a generic feature for this

---

Like this approach a lot.

I wonder if we should ask Replicated whether there is a built-in way to get the whole KOTS config as config map or if not, create a feature request. A template function that returns the whole KOTS config as YAML would probably be good enough.

I think, that creating a generic solution in KOTS should not be that hard for Replicated. It would make our lives much easier to maintain this.

What do you think?

_However, that does not block this pull request at all!_

Reply:
It is possible to get this as a configmap, which we indeed do in the [support bundle](https://github.com/gitpod-io/gitpod/blob/main/install/kots/manifests/kots-support-bundle.yaml#L109,L122).

The problem with this as an approach is the KOTS config will have many sensitive pieces of data (passwords for DB/Registry/Storage etc) which we don't actually need for the Installer, as these are sent into secrets that the Installer then references.

We could of course store this as a secret rather than a configmap, but this would still contain information that we don't actually need for the Installer. I do see the benefit to not having to manage this though, but this then has the danger of thinking we're exposing something that we're not and errors being caused.

I'd propose that, for this PR, we keep it as-is and continue to discuss this subject when doing the general refactoring that we're doing.

Reply:
> I'd propose that, for this PR, we keep it as-is and continue to discuss this subject when doing the general refactoring that we're doing.

100 % agree.